spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkChangelogScan.java

```diff
@@ -66,7 +66,8 @@ class SparkChangelogScan implements Scan, SupportsReportStatistics {
       IncrementalChangelogScan scan,
       SparkReadConf readConf,
       Schema expectedSchema,
-      List<Expression> filters) {
+      List<Expression> filters,
+      boolean emptyScan) {
```

Comment:
An alternative way is to pass a null scan object to avoid an additional parameter, but either works for me.